dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -406,10 +406,7 @@ public CloseableClient build() {
                     .setConnPoolPolicy(PoolReusePolicy.LIFO)
                     .setDefaultSocketConfig(SocketConfig.custom()
                             .setSoKeepAlive(true)
-                            // The default socket configuration socket timeout only applies prior to request execution.
-                            // By using the connect timeout here, we apply it to the handshake in addition to the
-                            // socket.connect call.
-                            .setSoTimeout(connectTimeout)
+                            .setSoTimeout(socketTimeout)
```

Comment:
slap a bit of a comment explaining that we explicitly do not want to consider handashake time towards the timeout